### PR TITLE
[#168726815] Invalidate old wallet token on new login

### PR DESCRIPTION
**Pull request by BurnedMarshal**

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a description of your PR changes.
  </td>
    </tr>
  </tbody>
</table>



## Affected stories

  * 🌟 [#168726815](https://www.pivotaltracker.com/story/show/168726815): Quando una sessione viene invalidata a causa di un nuovo login, devono essere invalidati i vecchi wallet_token

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/services/redisSessionStorage.ts [hunk 1]
```diff
@@ -430,8 +433,48 @@ export default class RedisSessionStorage extends RedisStorageUtils
...
+                return some(errorOrDeserializedUser.value.wallet_token);
+              } catch {
+                return none;
+              }
+            })
+        )
+        .filter(isSome)
+        .map(_ => `${walletKeyPrefix}${_.value}`);
+      // Delete all active session tokens and wallet tokens that are different
+      // from the new one genetated and provided inside user object.
```
Comment (gunzip) [inline src/services/redisSessionStorage.ts]:
```suggestion
      // from the new one generated and provided inside user object.
```

File: src/services/redisSessionStorage.ts [hunk 2]
```diff
@@ -430,8 +433,48 @@ export default class RedisSessionStorage extends RedisStorageUtils
...
+      // Map every user payload with an Option<WalletToken>
+      // If the value is invalid or must be skipped, it will be mapped with none
+      const walletsToken = errorOrUserWalletTokens
+        .fold(
+          _ => [],
+          _ =>
+            _.map(value => {
+              try {
+                // TODO: When https://github.com/teamdigitale/io-backend/pull/511 will be merged,
+                // update the following code using parseJSON ported from fp-ts
```
Comment (gunzip) [inline src/services/redisSessionStorage.ts]: it's merged

Comment (BurnedMarshal) [reply src/services/redisSessionStorage.ts]: Fixed in 323339b

File: src/services/redisSessionStorage.ts [hunk 3]
```diff
@@ -432,8 +423,52 @@ export default class RedisSessionStorage extends RedisStorageUtils
...
+        );
+      });
+      // Map every user payload with an Option<WalletToken>
+      // If the value is invalid or must be skipped, it will be mapped with none
+      const walletsToken = errorOrUserWalletTokens
+        .fold(
+          _ => [],
+          _ =>
+            _.map(value => {
+              try {
```
Comment (gunzip) [inline src/services/redisSessionStorage.ts]: is there some methods inside this try / catch that may throw ? (it looks like no method can actually throw). is fo, we can omit this try / catch part.

Comment (BurnedMarshal) [reply src/services/redisSessionStorage.ts]: Fixed in 60f3781
I've also removed code duplication.

File: src/services/redisSessionStorage.ts [hunk 4]
```diff
@@ -432,8 +417,40 @@ export default class RedisSessionStorage extends RedisStorageUtils
             _ !== `${sessionInfoKeyPrefix}${user.session_token}`
         )
         .map(_ => `${sessionKeyPrefix}${_.split(sessionInfoKeyPrefix)[1]}`);
+      /**
```
Comment (gunzip) [inline src/services/redisSessionStorage.ts]: please restrict this comment style to function description : )

Comment (BurnedMarshal) [reply src/services/redisSessionStorage.ts]: fixed in 276bafa

**Pull request merged**